src/Features/CSharp/Portable/UsePatternCombinators/CSharpUsePatternCombinatorsCodeFixProvider.cs

```diff
@@ -0,0 +1,108 @@
+
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(
+                new MyCodeAction(c => FixAsync(context.Document, context.Diagnostics.First(), c)),
+                context.Diagnostics);
+            return Task.CompletedTask;
+        }
+
+        protected override async Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.RequireSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            foreach (var diagnostic in diagnostics)
+            {
+                var location = diagnostic.Location;
+                var expression = editor.OriginalRoot.FindNode(location.SourceSpan, getInnermostNodeForTie: true);
+                var operation = semanticModel.GetOperation(expression, cancellationToken);
+                RoslynDebug.AssertNotNull(operation);
+                var pattern = CSharpUsePatternCombinatorsAnalyzer.Analyze(operation);
+                RoslynDebug.AssertNotNull(pattern);
+                var patternSyntax = AsPatternSyntax(pattern).WithAdditionalAnnotations(Formatter.Annotation);
+                editor.ReplaceNode(expression, IsPatternExpression((ExpressionSyntax)pattern.Target.Syntax, patternSyntax));
+            }
+        }
+
+        private static PatternSyntax AsPatternSyntax(AnalyzedPattern pattern)
+            => pattern switch
+            {
+                Binary p => BinaryPattern(
+                    p.IsDisjunctive ? SyntaxKind.OrPattern : SyntaxKind.AndPattern,
+                    AsPatternSyntax(p.Left).Parenthesize(),
+                    Token(p.Token.LeadingTrivia, p.IsDisjunctive ? SyntaxKind.OrKeyword : SyntaxKind.AndKeyword,
+                        TriviaList(p.Token.GetAllTrailingTrivia())),
+                    AsPatternSyntax(p.Right).Parenthesize()),
+                Constant p => ConstantPattern(p.ExpressionSyntax),
+                Source p => p.PatternSyntax,
+                Type p => TypePattern(p.TypeSyntax),
+                Relational p => RelationalPattern(Token(MapToSyntaxKind(p.OperatorKind)), p.Value.Parenthesize()),
+                Not p => UnaryPattern(AsPatternSyntax(p.Pattern).Parenthesize()),
+                var p => throw ExceptionUtilities.UnexpectedValue(p)
+            };
+
+        private class MyCodeAction : CodeAction.DocumentChangeAction
+        {
+            public MyCodeAction(Func<CancellationToken, Task<Document>> createChangedDocument)
+                : base(CSharpAnalyzersResources.Use_pattern_matching, createChangedDocument)
+            {
+            }
+
+            internal override CodeActionPriority Priority => CodeActionPriority.Low;
```

Comment:
doc.

Reply:
copied from another feature. should this be `low`?

Reply:
it should probably be low if the feature is set to hidden, and medium otherwise.    this oculd be very noisy and that worries me. how many hits did you find in roslyn alone out of curiosity?

Reply:
I'm not clear what priority this sets. Does it define how far up it shows up in the menu?

>how many hits did you find in roslyn alone

https://github.com/dotnet/roslyn/pull/43596#issuecomment-618403732 is just one project. so I assume a lot.